File: Embassat/View/Schedule/CADEMScheduleHeaderView.m
//
//  CADEMScheduleHeaderView.m
//  Embassat
//
//  Created by Joan Romano on 20/05/14.
//  Copyright (c) 2014 Crows And Dogs. All rights reserved.
//

#import "CADEMScheduleHeaderView.h"

#import "UIColor+EMAdditions.h"

@interface CADEMScheduleHeaderView ()

@property (nonatomic, weak) IBOutlet UIView *thursdayContainer, *fridayContainer, *saturdayContainer;
@property (nonatomic, weak) IBOutlet UILabel *thursdayLabel, *fridayLabel, *saturdayLabel;

@property (nonatomic, strong) RACSubject *daySelectedSignal;

@end

@implementation CADEMScheduleHeaderView

- (void)awakeFromNib
{
    [super awakeFromNib];

    self.thursdayLabel.textColor = [UIColor whiteColor];
    self.thursdayContainer.backgroundColor = [UIColor em_scheduleHeaderBackgroundColor];
    self.daySelectedSignal = [RACSubject subject];
    
    [@[self.thursdayContainer, self.fridayContainer, self.saturdayContainer] enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
        UITapGestureRecognizer *tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(containerTapped:)];
        [subview addGestureRecognizer:tapGesture];
    }];
    
    self.thursdayLabel.font = self.fridayLabel.font = self.saturdayLabel.font = [UIFont em_detailFontOfSize:15.0f];
}

- (void)containerTapped:(UITapGestureRecognizer *)sender
{
    NSArray *containers = @[self.thursdayContainer, self.fridayContainer, self.saturdayContainer];
    
    [[containers.rac_sequence filter:^BOOL(UIView *view) {
        return view.tag == sender.view.tag;
    }].array enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
        subview.backgroundColor = [UIColor em_scheduleHeaderBackgroundColor];
        ((UILabel *)[subview.subviews firstObject]).textColor = [UIColor whiteColor];
    }];
    
    [[containers.rac_sequence filter:^BOOL(UIView *view) {
        return view.tag != sender.view.tag;
    }].array enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
        subview.backgroundColor = [UIColor em_scheduleHeaderDeselectedBackgroundColor];
        ((UILabel *)[subview.subviews firstObject]).textColor = [UIColor em_scheduleHeaderDeselectedTextColor];
    }];
    
    [(RACSubject *)self.daySelectedSignal sendNext:@(sender.view.tag)];
}

@end

File: Embassat/View/Schedule/CADEMScheduleHeaderView.swift
//
//  CADEMScheduleHeaderView.swift
//  Embassa't
//
//  Created by Joan Romano on 25/05/15.
//  Copyright (c) 2015 Crows And Dogs. All rights reserved.
//

import UIKit

public class CADEMScheduleHeaderView: UICollectionReusableView {
    
    @IBOutlet weak var thursdayContainer: UIView?
    @IBOutlet weak var fridayContainer: UIView?
    @IBOutlet weak var saturdayContainer: UIView?
    @IBOutlet weak var thursdayLabel: UILabel?
    @IBOutlet weak var fridayLabel: UILabel?
    @IBOutlet weak var saturdayLabel: UILabel?
    
    public let daySelectedSignal: RACSubject = RACSubject()
    
    public override func awakeFromNib() {
        super.awakeFromNib()
        
        thursdayLabel?.textColor = UIColor.whiteColor()
        thursdayContainer?.backgroundColor = UIColor.em_scheduleHeaderBackgroundColor()
        
        let containers: Array<UIView?> = [thursdayContainer, fridayContainer, saturdayContainer]
        
        for view: UIView? in containers {
            let tapGesture = UITapGestureRecognizer(target: self, action: "containerTapped:")
            view?.addGestureRecognizer(tapGesture)
        }
        
        thursdayLabel?.font = UIFont.em_detailFontOfSize(15.0)
        fridayLabel?.font = UIFont.em_detailFontOfSize(15.0)
        saturdayLabel?.font = UIFont.em_detailFontOfSize(15.0)
    }
    
    func containerTapped(sender: UITapGestureRecognizer) {
        let containers: Array<UIView?> = [thursdayContainer, fridayContainer, saturdayContainer]
        
        for selectedView: UIView? in containers.filter({ (view: UIView?) -> Bool in
            return view?.tag == sender.view?.tag
        }) {
            selectedView?.backgroundColor = UIColor.em_scheduleHeaderBackgroundColor()
            
            if let label = selectedView?.subviews.first as? UILabel {
                label.textColor = UIColor.whiteColor()
            }
        }
        
        for unSelectedView: UIView? in containers.filter({ (view: UIView?) -> Bool in
            return view?.tag != sender.view?.tag
        }) {
            unSelectedView?.backgroundColor = UIColor.em_scheduleHeaderDeselectedBackgroundColor()
            
            if let label = unSelectedView?.subviews.first as? UILabel {
                label.textColor = UIColor.em_scheduleHeaderDeselectedTextColor()
            }
        }
        
        daySelectedSignal.sendNext(sender.view?.tag)
    }
}
